mlir/lib/Dialect/Linalg/Transforms/Transforms.cpp

```diff
@@ -1181,7 +1221,23 @@ LogicalResult GeneralizeOuterUnitDimsPackOpPattern::matchAndRewrite(
   SmallVector<int64_t> transpShape = readShape;
   applyPermutationToVector<int64_t>(transpShape, perm);
 
-  Value empty = rewriter.create<tensor::EmptyOp>(loc, transpShape, elemType);
+  // If there's a tile with a scalable size, retrieve its size. ATM only 1
+  // scalable tile is allowed.
+  Value scalableSize;
+  for (auto tile : packOp.getMixedTiles()) {
+    if (tile.is<Value>()) {
+      assert(!scalableSize && "Only one scalable size is supported ATM.");
+      scalableSize = cast<Value>(tile);
+      assert(vector::getConstantVscaleMultiplier(scalableSize) &&
+             "This dynamic shape is not a multiple of vscale!");
+    }
+  }
+
+  Value empty =
+      ShapedType::isDynamicShape(cast<ShapedType>(input.getType()).getShape())
+          ? rewriter.create<tensor::EmptyOp>(loc, transpShape, elemType,
+                                             scalableSize)
+          : rewriter.create<tensor::EmptyOp>(loc, transpShape, elemType);
```

Comment:
You're probably right that this can be simplified, but it's not immediately obvious to me :) Note that for this example:
```mlir
%0 = tensor.pack %arg0 inner_dims_pos = [3, 2] inner_tiles = [8, 32] into %arg1 : tensor<1x1x32x8xf32> -> tensor<1x1x1x1x8x32xf32>
```

`readSizes` and `readShape` are `[1, 1, 32, 8]` and `[32, 8]`, respectively. I need to take another look 😅

Reply:
oh I see, I misread the code. The unit dims are not inserted to `readShape`. Perhaps we can add `SmallVector<OpFoldResult> transpShape` to l.1190, and insert the `OpFoldResult` values to the vector and use it here.

Reply:
Updated. There's quite a few vectors with shapes/offsets/etc. I've renamed two of them so that we have e.g. `readShapeForExtractSlice` rather than a bit abstract `readShape`. Let me know what you think :)